File: heex_ros/Heex_SDK_2_39_1/get-started/python/getStartedMonitor.py
#!/usr/bin/env python3
import sys
import os

# Adding the PyHeexCustomerSide Lib to PYTHONPATH. Depending on your folder structure, you might need to update this block
sys.path.insert(1, "../sdk/lib")
sys.path.insert(1, "../../sdk/lib")
sys.path.insert(1, "../lib")
from PyHeexCustomerSide import BooleanMonitor


class GetStartedMonitor(BooleanMonitor):
    def __init__(self, uuid, serverIp, serverPort):
        BooleanMonitor.__init__(self, uuid, serverIp, serverPort)


if __name__ == "__main__":
    try:
        # TODO: Replace this left Monitor UUID with the one provided (by the Web Platform) for your Trigger
        monitorUuid = "D-d6fa84b8-26cb-44b6-9bde-89ec3ef5b4d9(1.0.0)"
        if len(sys.argv) >= 2:
            monitorUuid = sys.argv[1]

        #
        # Configure the Monitor
        #
        serverIp = "127.0.0.1"
        serverPort = 4242
        print("Monitor_uuid = ", monitorUuid)
        print("serverIp = ", serverIp)
        print("serverPort = ", serverPort)

        #
        # Launch Monitor
        #
        getStartedMonitor = GetStartedMonitor(monitorUuid, serverIp, serverPort)
        getStartedMonitor.awaitReady()
        getStartedMonitor.updateValue(True, getStartedMonitor.getTimestampStr())

    except Exception as e:
        print("Error", e)
        exit(1)

File: heex_ros/Heex_SDK_2_39_1/get-started/cpp/getStartedMonitor.cpp
#include "getStartedMonitor.h"

#include <iostream>
#include <thread>

Heex::GetStarted::GetStartedMonitor::GetStartedMonitor(const std::string& uuid, const std::string& serverIp, unsigned int& serverPort) : BooleanMonitor(uuid, serverIp, serverPort)
{
  HEEX_LOG(info) << "GetStartedMonitor | " << uuid << " " << serverIp << " " << serverPort << " " << std::endl;
}

int main(int argc, char* argv[])
{
  // TODO: Replace this left Monitor UUID with the one provided (by the Web Platform) for your Trigger
  std::string monitorUuid = "D-d6fa84b8-26cb-44b6-9bde-89ec3ef5b4d9(1.0.0)";
  if (argc > 1)
  {
    monitorUuid = argv[1]; // Parse the UUID as an optional parameter.
  }

  //
  // Configure the Monitor
  //
  const std::string serverIp = "127.0.0.1";
  unsigned int serverPort    = 4242;
  Heex::GetStarted::GetStartedMonitor myMonitor(monitorUuid, serverIp, serverPort);

  //
  // Launch Monitor
  //
  myMonitor.awaitReady();
  myMonitor.updateValue(true);

  return 0;
}
